### ISSUE TITLE
On Smart Devices (Android, iOS), showing the controls bar always pauses the playback

### ISSUE THREAD

**Issue by gabrieletassoni** [openplayerjs/openplayerjs#325]
Using PCs, with a mouse, the hover effect which raises the controls bar is ok. On Smartphones, in which there is no mouse, but anything is made by tapping the screen directly, tapping on the video to show the controls bar, just to raise volume or go fullscreen, makes the playback pause, quite annoying.
Is there a way to have a first tap on the player, which just raises the controls bar and won't pause the playback? A second tap on the player would pause it, but if one wants to just tap on fullscreen, as his second tap, the playback won't be affected.

Thank you very much

**Comment by rafa8626** [comment]
I could make it work that way; it may require some refinement in the layer that shows the play/pause button, but I think it’s feasible

**Comment by rafa8626** [comment]
@gabrieletassoni I’ve included this fix in the v3.0.0 branch; can you test it and let me know if it worked for you?

**Comment by gabrieletassoni** [comment]
Yes, I would test it in the next hours, I'll let you know, many thanks ^_^

**Comment by rafa8626** [comment]
@gabrieletassoni I wanted to check if the branch worked for you; if so, I’ll prepare a release with it, but let me know

**Comment by rafa8626** [comment]
@gabrieletassoni I’ll check that because that doesn’t sound like a behavior that was there before

**Comment by gabrieletassoni** [comment]
Yes, I do it on Monday, since i don't have my dev PC with me.
By now I just can share a link to the previous behaviour: it doesn't have the tap to show controls behaviour, but I can clearly see that the controls appear above the video even when in Fullscreen: https://www.alchemic.it/vod.html
On Monday I'll post a link to another instance, with openplayerjs V3, in which the controls won't show when tapping the video while in Fullscreen.

**Comment by rafa8626** [comment]
@gabrieletassoni can you post the link using the v3.0.0 branch to see the issue you described there, please?

**Comment by gabrieletassoni** [comment]
Hi, here it is: https://alchemic.it/vod-test.html
As soon as I tap, using my android 10 smartphone, on the fullscreen button and the player goes fullscreen, the controls are lost and tapping again on the video won't bring them up again, until I manage to exit fullscreen mode (usually by making the android back button show up and use it).

**Comment by rafa8626** [comment]
@gabrieletassoni I’ll need to figure out how to test Android since I don’t own a device or have an emulator for it

**Comment by gabrieletassoni** [comment]
Some prebuilt images for Virtualbox or vmware can be found here: https://www.osboxes.org/android-x86/
I tested the https://alchemic.it/vod-test.html link on a iOS device, and there it brings up the controls for the video, but in a strange way, they are not the openplayerjs (red-colored) controls they look like native controls.
Moreover, on my iPad it disappeared the fix for which the first tap on the video would just bring up controls and not pause the playback. On the android smartphone, the same page works as expected.

**Comment by rafa8626** [comment]
@gabrieletassoni in an iPhone that’s expected behavior because it uses the built-in player when you go fullscreen. So, the only issue is with iPad then?

**Comment by rafa8626** [comment]
@gabrieletassoni I fixed the issue with iPad; please check it and let me know if it worked for you

**Comment by gabrieletassoni** [comment]
Hi, I'll check for the ipad behaviour, but here I recorded a comparison on android for the other problem (fullscreen without controls), as you can see from attached video, the version 2 of openplayerjs, while fullscreen, shows the controls by tapping the screen, with v3, this behaviour is no more there.

Thanks again ^_^

https://user-images.githubusercontent.com/11076471/154450069-dbbc7a3f-8534-40da-9174-082ec6fbdd79.mp4

**Comment by rafa8626** [comment]
@gabrieletassoni I'll check this; it's weird since I didn't change anything in regards of that behavior with the fullscreen in the new branch

**Comment by rafa8626** [comment]
@gabrieletassoni I think I got it, although I still don't understand how that worked in 2.0 but not in 3.0. Please download the new files under the `dist` folder in the `v3.0.0` branch and let me know if it worked for you as expected

**Comment by rafa8626** [comment]
@gabrieletassoni I may need to revert this fix, because it an iPhone, it doesn't do what it is supposed to. I'll continue investigating

**Comment by rafa8626** [comment]
@gabrieletassoni I got it; there was a small mismatch between what should have been the proper target to go fullscreen what caused this issue. It should be good now; just download the dist folder from the v3.0.0 branch and test them once more. It should be good for all devices now

**Comment by gabrieletassoni** [comment]
Yes, it works all as expected

**Comment by rafa8626** [comment]
Thanks for testing this; I’ll deploy this fix next week
